datastore/datastore.go

```diff
@@ -650,18 +657,99 @@ func (s *FileStore) GetAll(ctx context.Context, query Query, dst interface{}) ([
 			return nil, err
 		}
 		// The underlying entity type is a pointer, so we need its indirect type.
+
+		// Apply field filters if needed.
+		if len(q.fieldFilters) > 0 && !matchesFieldFilters(e, q.fieldFilters) {
+			continue
+		}
+
 		ev := reflect.Indirect(reflect.ValueOf(e))
 		// As with the Cloud datastore, if the Key field is present we populate it.
 		fld := ev.FieldByName("Key")
 		if fld.IsValid() {
 			fld.Set(reflect.ValueOf(k))
 		}
+
 		dv.Set(reflect.Append(dv, ev))
 	}
 
 	return keys, nil
 }
 
+// matchesFieldFilters returns true if the given entity matches all the specified fieldFilters.
+// It uses reflection to extract struct fields by name and compares their values based on the
+// provided operator (e.g., "=", "<=", ">"). If any filter does not match, it returns false.
+func matchesFieldFilters(e Entity, filters []fieldFilter) bool {
+	v := reflect.Indirect(reflect.ValueOf(e))
+	for _, f := range filters {
+		field := v.FieldByName(f.Field)
+		if !field.IsValid() {
+			return false
+		}
+
+		switch f.Operator {
+		case "=":
+			if field.Interface() != f.Value {
+				return false
+			}
+		case "<":
```

Comment:
You can make this more readable without loss of efficiency by grouping all of the non-equality operators, e.g.,

```
case "<", ">", "<=", ">=":
    if !compare(field.Interface(), f.Value, f.Operator) {
        return false
    }
```